cpp/models/abm/location.h

```diff
@@ -96,48 +95,26 @@ struct Cell {
 class Location
 {
 public:
-    /**
-     * @brief Construct a Location of a certain LocationId.
-     * @param[in] loc_id The #LocationId.
-     * @param[in] num_agegroups [Default: 1] The number of age groups in the model.
-     * @param[in] num_cells [Default: 1] The number of Cell%s in which the Location is divided.
-     */
-    Location(LocationId loc_id, size_t num_agegroups = 1, uint32_t num_cells = 1);
-
     /**
      * @brief Construct a Location with provided parameters. 
      * @param[in] loc_type The #LocationType.
-     * @param[in] index The index of the Location.
+     * @param[in] loc_id The index of the Location in the World.
```

Comment:
```suggestion
     * @param[in] loc_id The LocationId of the Location in the World.
```

Reply:
I also suggested to start using the name 'Model' instead of World.

Reply:
"The index ..." does give you additional information on what loc_id is, while "The LocationId ..." just names the type again, are you sure this should be replaced?

And while the rename World to Model is planned, it is out of scope for this issue, and I will not partially rename it now.

Reply:
That's true. I guess "index" or "Unique identifier for the Location in the World" could also work. However, please be consistent with the naming in all comments.

Reply:
What do you mean by consistent naming? Do you want all LocationId params to have the exact same doc comment?

Not sure that is necessary, but what about "Unique identifier for the Location in a World. Mostly obtained from World::add_location.".

Reply:
> What do you mean by consistent naming? Do you want all LocationId params to have the exact same doc comment?
Not exactly, just sometimes loc_id is preferred as "Id of the Location", sometimes as "Index of the Location", or "location's identifier".  Maybe choose one of "identifier" or "index"?

Reply:
To be honest, all of those are correct, and we state e.g. "The assigned Location is saved by the index of its LocationId", so it is not a secret either. ID is also just a shorthand for identifier.

More importantly, my changes here do not cover several instances of uses of id/index, so if you really want a unified description, you should open a new issue for this.

Reply:
Ok, I guess we can leave them now and mention it in a future issue when all the documentation is updated.